pkg/observable/channel/observable.go

```diff
@@ -63,6 +64,14 @@ func WithPublisher[V any](publishCh chan V) option[V] {
 	}
 }
 
+func (obsvbl *channelObservable[V]) Next(ctx context.Context) V {
```

Comment:
Can you add a godoc on what this is or why its needed?

Reply:
Added a godoc comment. It's not strictly necessary but will be convenient in forthcoming PRs.

Reply:
`It's not strictly necessary but will be convenient in forthcoming PRs.`

I'm a bit confused as to when we should or shouldn't use it. #PUC when you have time, but not a blocker in this PR.